go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -4742,35 +4742,40 @@ func (s *VtctldServer) ValidateKeyspace(ctx context.Context, req *vtctldatapb.Va
 		return resp, err
 	}
 
-	resp.ResultsByShard = make(map[string]*vtctldatapb.ValidateShardResponse, len(shards))
+	if len(shards) == 0 {
+		resp.Results = append(resp.Results, fmt.Sprintf("no shards found in keyspace %v", req.Keyspace))
```

Comment:
can we just `return resp, err` here, skip the `else` clause and unindent back the rest of the code?